### PR TITLE
Include config option 'unauthenticated_metrics_access'

### PR DESCRIPTION
**Pull request by konradasb**
This PR adds option for unauthenticated metrics access to `/v1/sys/metrics`
https://www.vaultproject.io/docs/configuration/listener/tcp#unauthenticated_metrics_access

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: metadata.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@
 issues_url        'https://github.com/sous-chefs/vault/issues'
 source_url        'https://github.com/sous-chefs/vault'
 chef_version      '>= 13'
-version           '4.2.0'
+version           '4.2.1'
```
Comment (ramereth) [inline metadata.rb:9]: Please don't bump the version in PRs. We'll do that separately.

**Review by ramereth — CHANGES_REQUESTED** [review]
Can you please also handle the issues mentioned by the @kitchen-porter? Thank you for the contribution!

**Comment by konradasb** [comment]
Hi, thanks for review, updated the PR :)

**Review by ramereth — APPROVED** [review]

**Pull request merged**